src/batched/KokkosBatched_InverseLU_Serial_Impl.hpp

```diff
@@ -0,0 +1,196 @@
+            }
+        }
+        else { //WViewType::rank = 2
+            B_stride_0 = A.stride_0();
+            B_stride_1 = A.stride_1();
+        }
+
+        auto B = Kokkos::View<ScalarType**, Kokkos::LayoutStride, typename WViewType::memory_space, Kokkos::MemoryTraits<Kokkos::Unmanaged> >(W.data(), Kokkos::LayoutStride(A.extent(0), B_stride_0, A.extent(1), B_stride_1));
+
+        const ScalarType one(1.0);
+        
+#if defined(KOKKOS_ENABLE_PRAGMA_UNROLL)
+#pragma unroll
+#endif
+        for (int i=0;i<A.extent(1);++i) {
+            B(i,i) = one;
+        }
+
+        //First, compute L inverse by solving the system L*Linv = I for Linv
+        SerialTrsm<Side::Left,Uplo::Lower,Trans::NoTranspose,Diag::Unit,Algo::Trsm::Unblocked>::invoke(one, A, B);
+        //Second, compute A inverse by solving the system U*Ainv = Linv for Ainv
+        SerialTrsm<Side::Left,Uplo::Upper,Trans::NoTranspose,Diag::NonUnit,Algo::Trsm::Unblocked>::invoke(one, A, B);
+
+#if defined(KOKKOS_ENABLE_PRAGMA_UNROLL)
+#pragma unroll
+#endif
+         for (int i=0;i<A.extent(0);++i)
+            for (int j=0;j<A.extent(1);++j)
+                A(i,j) = B(i,j);
+
+        return 0;
+    }
+    
+    template<>
+    template<typename AViewType,
+             typename WViewType>
+    KOKKOS_INLINE_FUNCTION
+    int
+    SerialInverseLU<Algo::InverseLU::Blocked>::
+    invoke(const AViewType &A,
+           const WViewType &W) {
+        static_assert(AViewType::rank == 2, "A should have two dimensions");
+        static_assert(std::is_same<typename AViewType::memory_space, typename WViewType::memory_space>::value, "A and W should be on the same memory space");
+        assert(A.span()*sizeof(typename AViewType::value_type) <= W.span()*sizeof(typename WViewType::value_type));
+
+        typedef typename AViewType::value_type ScalarType;
+
+        int B_stride_0, B_stride_1;
+        if (WViewType::rank == 1) {
+            if (std::is_same<typename WViewType::array_layout, Kokkos::LayoutRight>::value){
+                B_stride_0 = A.extent(1);
+                B_stride_1 = 1;
+            } 
+            else {
+                B_stride_0 = A.stride_0();
+                B_stride_1 = A.stride_0()*A.extent(0);
+            }
+        }
+        else { //WViewType::rank = 2
+            B_stride_0 = A.stride_0();
+            B_stride_1 = A.stride_1();
+        }
+
+        auto B = Kokkos::View<ScalarType**, Kokkos::LayoutStride, typename WViewType::memory_space, Kokkos::MemoryTraits<Kokkos::Unmanaged> >(W.data(), Kokkos::LayoutStride(A.extent(0), B_stride_0, A.extent(1), B_stride_1));
+
+        const ScalarType one(1.0);
+
+#if defined(KOKKOS_ENABLE_PRAGMA_UNROLL)
+#pragma unroll
+#endif
+        for (int i=0;i<A.extent(1);++i) {
+            B(i,i) = one;
+        }
+
+        //First, compute L inverse by solving the system L*Linv = I for Linv
+        SerialTrsm<Side::Left,Uplo::Lower,Trans::NoTranspose,Diag::Unit,Algo::Trsm::Blocked>::invoke(one, A, B);
```

Comment:
Plese use the internal interface instead of the front interface which includes unpacking meta data of View A and B.

Reply:
I tried internal interfaces but there is a weird Intel compiler bug that gives wrong results for the test case with double complex and unblocked algorithm type. So, use the front interfaces instead for now.